src/Avalonia.Diagnostics/Diagnostics/Controls/VirtualizedTreeView/VirtualizedTreeView.axaml

```diff
@@ -0,0 +1,30 @@
+<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:treeView="clr-namespace:Avalonia.Diagnostics.Controls.VirtualizedTreeView"
+        x:ClassModifier="internal">
+  <Styles.Resources>
+    <ControlTheme x:Key="{x:Type treeView:VirtualizedTreeView}"
+                  TargetType="treeView:VirtualizedTreeView">
+      <Setter Property="Template">
+        <ControlTemplate>
+          <treeView:VirtualizedTreeListBox ItemsSource="{TemplateBinding Source}"
+                                           SelectedValueBinding="{ReflectionBinding Node}"
+                                           SelectedValue="{TemplateBinding SelectedItem, Mode=TwoWay}">
+            <treeView:VirtualizedTreeListBox.ItemTemplate>
+              <DataTemplate x:DataType="treeView:FlatTreeNode">
+                <ContentControl Content="{CompiledBinding Node}"
+                                ContentTemplate="{Binding $parent[treeView:VirtualizedTreeView].ItemTemplate}" />
+              </DataTemplate>
+            </treeView:VirtualizedTreeListBox.ItemTemplate>
+          </treeView:VirtualizedTreeListBox>
+        </ControlTemplate>
+      </Setter>
+    </ControlTheme>
+  </Styles.Resources>
+
+  <Style Selector="treeView|VirtualizedTreeView treeView|VirtualizedTreeViewItem">
+    <Setter Property="IsExpanded" Value="{CompiledBinding Node.IsExpanded}" x:DataType="treeView:FlatTreeNode" />
+    <Setter Property="HasChildren" Value="{CompiledBinding Node.HasChildren}" x:DataType="treeView:FlatTreeNode" />
+    <Setter Property="IndentLevel" Value="{CompiledBinding Level}" x:DataType="treeView:FlatTreeNode" />
+  </Style>
```

Comment:
Can be replaced with something like
```xaml
    <ControlTheme x:Key="{x:Type treeView:VirtualizedTreeView}"
                  TargetType="treeView:VirtualizedTreeView">
           <Setter Property="ItemContainerTheme">
                <ControlTheme BasedOn="{x:Type treeView:VirtualizedTreeViewItem}"
                                          TargetType="treeView:VirtualizedTreeViewItem">
                         <Setter Property="IsExpanded" Value="{CompiledBinding Node.IsExpanded}" x:DataType="treeView:FlatTreeNode" />
                         <Setter Property="HasChildren" Value="{CompiledBinding Node.HasChildren}" x:DataType="treeView:FlatTreeNode" />
                         <Setter Property="IndentLevel" Value="{CompiledBinding Level}" x:DataType="treeView:FlatTreeNode" />
```

Reply:
I.e. avoiding these free-loading styles in the end

Reply:
Still learning how to use ControlThemes :)) This makes sense! Fixed

Reply:
Oh, I was thinking it's a control theme for ItemsControl/ListBox, which had existing ItemContainerTheme property, but you had to add new property in your case. Is it passed down to the ListBox?

Reply:
I have added `ItemContainerTheme="{TemplateBinding ItemContainerTheme}"` to the ListBox in ControlTemplate and it seems to work just fine.

Reply:
Good